test/state/state.cpp

```diff
@@ -53,24 +53,34 @@ int64_t compute_tx_intrinsic_cost(evmc_revision rev, const Transaction& tx) noex
 }
 
 /// Validates transaction and computes its execution gas limit (the amount of gas provided to EVM).
-/// @return  Non-negative execution gas limit for valid transaction
-///          or negative value for invalid transaction.
+/// @return  Execution gas limit or transaction validation error.
 std::variant<int64_t, std::error_code> validate_transaction(const Account& sender_acc,
     const BlockInfo& block, const Transaction& tx, evmc_revision rev) noexcept
 {
-    if (rev < EVMC_LONDON && tx.kind == Transaction::Kind::eip1559)
-        return make_error_code(TX_TYPE_NOT_SUPPORTED);
+    switch (tx.kind)
+    {
+    case Transaction::Kind::eip1559:
+        if (rev < EVMC_LONDON)
+            return make_error_code(TX_TYPE_NOT_SUPPORTED);
+
+        if (tx.max_priority_gas_price > tx.max_gas_price)
+            return make_error_code(TIP_GT_FEE_CAP);  // Priority gas price is too high.
+        [[fallthrough]];
 
-    if (rev < EVMC_BERLIN && tx.kind == Transaction::Kind::eip2930)
-        return make_error_code(TX_TYPE_NOT_SUPPORTED);
+    case Transaction::Kind::eip2930:
+        if (rev < EVMC_BERLIN)
+            return make_error_code(TX_TYPE_NOT_SUPPORTED);
+        [[fallthrough]];
+
+    case Transaction::Kind::legacy:;
+    }
 
-    if (tx.max_priority_gas_price > tx.max_gas_price)
-        return make_error_code(TIP_GT_FEE_CAP);  // Priority gas price is too high.
+    assert(tx.max_priority_gas_price <= tx.max_gas_price);
```

Comment:
So for `access_list` and `legacy` this can fail assertion now for some invalid input? I think we should still return an error.

Reply:
The `access_list` and `legacy` only have "gas price" and we set `max_priority_gas_price` and `max_gas_price` to the same value. That's why the assert.